Rubrik policy check and logging fixes

claytonpeters opened the pull request:
Fixed an issue where VMs that hadn't made it in to Rubrik yet were causing exceptions during the Rubrik policy check. This also highlighted an issue regardling logging in the corpus.Rubrik class, which is fixed by making TaskHelper and app look more similar by making TaskHelper provide a logger variable that is a cut-down version of app.logger

jakemalley commented:
Would it not be better to use the `logging` module instead of wrapping `print` with `TaskHelperLogger`. 

Then just set the logger object in helper to something similar to:
```
logger = logging.getLogger()
logger.setLevel(logging.INFO)
stdout_handler = logging.StreamHandler(sys.stdout)
stdout_handler.setFormatter(formatter)
logger.addHandler(stdout_handler)
```

or use the syslog or filehandler logger so we can actually get some nicer logs from `neocortex`.

claytonpeters commented:
I did consider that, but there was a reason I avoided that. That said, I was **very** tired when I wrote this yesterday - so, I'll have a rethink and might change that.

claytonpeters commented:
Right, I've reworked this, as well as all of the logging in NeoCortex so it is now handled as a logging object. I've removed all references to the syslog module.

jakemalley commented:
This looks good to merge - lets do it.
